core/src/main/resources/defaults/blueprints/cdp-data-engineering-710.bp

```diff
@@ -80,7 +84,7 @@
               },
               {
                 "name": "resourcemanager_capacity_scheduler_configuration",
-                "value": "<configuration><property><name>yarn.scheduler.capacity.root.queues</name><value>default</value></property><property><name>yarn.scheduler.capacity.root.capacity</name><value>100</value></property><property><name>yarn.scheduler.capacity.root.default.capacity</name><value>100</value></property><property><name>yarn.scheduler.capacity.root.acl_submit_applications</name><value> </value></property><property><name>yarn.scheduler.capacity.root.acl_administer_queue</name><value> </value></property><property><name>yarn.scheduler.capacity.root.default.acl_submit_applications</name><value>*</value></property><property><name>yarn.scheduler.capacity.root.default.minimum-user-limit-percent</name><value>100</value></property><property><name>yarn.scheduler.capacity.maximum-am-resource-percent</name><value>0.33</value></property><property><name>yarn.scheduler.capacity.node-locality-delay</name><value>0</value></property><property><name>yarn.scheduler.capacity.schedule-asynchronously.maximum-threads</name><value>1</value></property><property><name>yarn.scheduler.capacity.schedule-asynchronously.scheduling-interval-ms</name><value>10</value></property><property><name>yarn.scheduler.capacity.per-node-heartbeat.maximum-offswitch-assignments</name><value>4</value></property></configuration>"
+                "value": "<configuration><property><name>yarn.scheduler.capacity.root.queues</name><value>default</value></property><property><name>yarn.scheduler.capacity.root.capacity</name><value>100</value></property><property><name>yarn.scheduler.capacity.root.default.capacity</name><value>100</value></property><property><name>yarn.scheduler.capacity.root.acl_submit_applications</name><value> </value></property><property><name>yarn.scheduler.capacity.root.acl_administer_queue</name><value> </value></property><property><name>yarn.scheduler.capacity.root.default.acl_submit_applications</name><value>*</value></property><property><name>yarn.scheduler.capacity.root.default.minimum-user-limit-percent</name><value>100</value></property><property><name>yarn.scheduler.capacity.maximum-am-resource-percent</name><value>0.33</value></property><property><name>yarn.scheduler.capacity.node-locality-delay</name><value>0</value></property><property><name>yarn.scheduler.capacity.schedule-asynchronously.maximum-threads</name><value>1</value></property><property><name>yarn.scheduler.capacity.schedule-asynchronously.scheduling-interval-ms</name><value>10</value></property></configuration>"
```

Comment:
Is the default value for off-switch assignments 1?

The YARN team is going to be putting in a patch which will help with this soon.

Reply:
> There's a few other configs that are in your doc / wiki.
> 
> hiveserver2_mv_files_thread
> hiveserver2_load_dynamic_partitions_thread_count
> hiveserver2_vectorized_adaptor_usage_mode=all
> 
> Do any of these need to be set?


@sidseth 
We definitely want to tune hiveserver2_mv_files_thread and hiveserver2_load_dynamic_partitions_thread_count but they require dynamic values depending upon the type of machine instance, this cannot be done through cluster template, hence omitted. More details about them are in the [wiki](https://cloudera.atlassian.net/wiki/spaces/ENG/pages/192414479/Config+tunings+for+Datahub+DE+cluster#3.-Hive-Tunings) 

hiveserver2_vectorized_adaptor_usage_mode=all, @rbalamohan said this is applicable only for Experian case, hence I did not change this. Experian can change this in their custom templates.



> Is the default value for off-switch assignments 1?
> 
> The YARN team is going to be putting in a patch which will help with this soon.

As far as I know default value of off-switch assignments is 1 thats why I removed this config from cluster template.

Reply:
Technically, tez container size, YARN container size are also dynamic.

Are the defaults for the thread configs good enough?, or should better defaults be set up in the template.

I'm fine with the patch going in as is. If we can make defaults a little better - great. Eventually, and hopefully, soon - all of this gets removed from the template.

Reply:
@sidseth 
After discussing with Rajesh I have updated defaults for hive.mv.files.thread & hive.load.dynamic.partitions.thread to 20.